src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -239,59 +239,59 @@ public static class Avx
         /// __int8 _mm256_extract_epi8 (__m256i a, const int index)
         ///   HELPER
         /// </summary>
-        public static sbyte Extract(Vector256<sbyte> value, byte index)
+        public static int Extract(Vector256<sbyte> value, byte index)
         {
             unsafe
             {
                 index &= 0x1F;
                 sbyte* buffer = stackalloc sbyte[32];
                 Store(buffer, value);
-                return buffer[index];
+                return (int)buffer[index];
```

Comment:
Question for future: Is this more efficient than a GetUpper and Extract combo?

Reply:
No, it's not. Here managed implementation is just the non-const fallback that simplifies JIT implementation of the imm-helpers.

Reply:
Can this be just `return ((sbyte*)(&value))[index];` ?

Reply:
No, you can't take the address of a generic.

Reply:
We could probably do a one line with `S.R.CS.Unsafe`

Reply:
`return Unsafe.Add<ref Unsafe.As<Vector256<sbyte>, sbyte>(ref value), index & 0x1F);`